### PR TITLE
support legacy global overrides from ~/.aws/taskcat_global_override.json

### LINKED ISSUE #357: Global override in ~/.aws is not honored
The global override (in `~/.aws/taskcat_global_override.json`) is currently not honored in v0.9.0; This is not intentional, and we intended to maintain backwards compatibility - for now. 

The correct approach is to honor it for the time being, and notify the user that this file will be depreciated in the future.

### LINKED ISSUE #359: Project overrides are based on cwd, not project root
https://github.com/aws-quickstart/taskcat/blob/c48e4edf7bf5ca54229542e1e89fb0420981455d/taskcat/_config.py#L86

Just need to add logic around the project_root, not cwd; plus tests. 

This should read

`legacy_override = Path( project_root / ci/taskcat_project_override.json” ).resolve()`

### PR DESCRIPTION
**Pull request by jaymccon**
## Overview

Fixes #357
Fixes #359

### CODE REVIEW

File: taskcat/_legacy_config.py [hunk 1]
```diff
@@ -89,3 +89,30 @@ def parse_legacy_config(project_root: Path):
...
+            LOG.warning(
+                f"Converting overrides to new format and saving in " f"{overrides_path}"
+            )
+            with open(str(overrides_path), "w") as file_handle:
+                file_handle.write(yaml.dump(override_params, default_flow_style=False))
+        else:
+            LOG.warning(
+                f"Ignoring legacy overrides as a current format "
+                f"file has been found in {str(overrides_path)}"
+            )
```
Comment (andrew-glenn) [inline taskcat/_legacy_config.py]: Where's the `return` statement?

Comment (jaymccon) [reply taskcat/_legacy_config.py]: err, this function doesn't return anything...

Comment (andrew-glenn) [reply taskcat/_legacy_config.py]: You're right, and it doesn't need to. Good to go.

**Pull request merged**